__mocks__/vscode.ts

```diff
@@ -0,0 +1,134 @@
+// Jtest Mock from https://github.com/jest-community/vscode-jest/blob/1867bde9ebf8c21d45409e7ad8a95a743ac4390d/__mocks__/vscode.ts
+// under MIT License
+
+const languages = {
+    createDiagnosticCollection: jest.fn(),
+    registerCodeLensProvider: jest.fn(),
+  };
+
+  const StatusBarAlignment = { Left: 1, Right: 2 };
+
+  const window = {
+    createStatusBarItem: jest.fn(() => ({
+      show: jest.fn(),
+      hide: jest.fn(),
+      tooltip: jest.fn(),
+    })),
+    showErrorMessage: jest.fn(),
+    showWarningMessage: jest.fn(),
+    createTextEditorDecorationType: jest.fn(),
+    createOutputChannel: jest.fn(),
+    showWorkspaceFolderPick: jest.fn(),
+    showQuickPick: jest.fn(),
+    onDidChangeActiveTextEditor: jest.fn(),
+    showInformationMessage: jest.fn(),
+    createWebviewPanel: jest.fn(),
+  };
+
+  const workspace = {
+    getConfiguration: jest.fn(),
+    workspaceFolders: [],
+    getWorkspaceFolder: jest.fn(),
+
+    onDidChangeConfiguration: jest.fn(),
+    onDidChangeTextDocument: jest.fn(),
+    onDidChangeWorkspaceFolders: jest.fn(),
+    onDidCreateFiles: jest.fn(),
+    onDidDeleteFiles: jest.fn(),
+    onDidRenameFiles: jest.fn(),
+    onDidSaveTextDocument: jest.fn(),
+    onWillSaveTextDocument: jest.fn(),
+  };
+
+  const OverviewRulerLane = {
+    Left: null,
+  };
+
+  const Uri = {
+    file: (f: any) => f,
+    parse: jest.fn(),
+    joinPath: jest.fn(),
+  };
+  const Range = jest.fn();
+  const Location = jest.fn();
+  const Position = jest.fn();
+  const Diagnostic = jest.fn();
+  const ThemeIcon = jest.fn();
+  const DiagnosticSeverity = { Error: 0, Warning: 1, Information: 2, Hint: 3 };
+  const ConfigurationTarget = { Global: 1, Workspace: 2, WorkspaceFolder: 3 };
+
+  const debug = {
+    onDidTerminateDebugSession: jest.fn(),
+    startDebugging: jest.fn(),
+    registerDebugConfigurationProvider: jest.fn(),
+  };
+
+  const commands = {
+    executeCommand: jest.fn(),
+    registerCommand: jest.fn(),
+    registerTextEditorCommand: jest.fn(),
+  };
+
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  const CodeLens = function CodeLens() {};
+
+  const QuickInputButtons = {
+    Back: {},
+  };
+
+  const tests = {
+    createTestController: jest.fn(),
+  };
+
+  const TestRunProfileKind = {
+    Run: 1,
+    Debug: 2,
+    Coverage: 3,
+  };
+  const ViewColumn = {
+    One: 1,
+    Tow: 2,
+  };
+
+  const TestMessage = jest.fn();
+  const TestRunRequest = jest.fn();
+  const ThemeColor = jest.fn();
+
+  const EventEmitter = jest.fn().mockImplementation(() => {
+    return {
+      fire: jest.fn(),
+    };
+  });
+
+  const QuickPickItemKind = {
+    Separator: -1,
+    Default: 0,
+  };
+
+  export = {
```

Comment:
On my side typescript complains it is using the CommonJS syntax, while the EcmaScript syntaxt should be used (`export default`). Yet it won't compile if I make that change.

I am not sure where is exactly the problem, but I noticed .eslintrc.js is also using a commonjs export. I also noticed you put this file into "ignorePatterns". I am thinking this was an attempt to solve that problem.

Reply:
This file comes from another repo and works fine. As long as we don't change it, it makes no sense to me to test it's linting, which follows a different standard in another project. Also it makes it easy to update this file in the future when the vscode API changes/adds new features. I suggest to ignore linting this file and keep it as original. A comment at the top clearly indicates where it comes from.

Reply:
Makes sense. However I am surprised it works. Both syntaxes are not supposed to be interchangeable within a project. As a proof, this file does not compile if you use the ecmascript syntax. It makes me think the "root project" is not configured properly.